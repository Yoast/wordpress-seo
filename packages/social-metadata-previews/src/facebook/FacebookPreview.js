/* External dependencies */
import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

/* Internal dependencies */
import FacebookSiteAndAuthorNames from "./FacebookSiteAndAuthorNames";
import FacebookImage from "./FacebookImage";
import FacebookTitle from "./FacebookTitle";
import FacebookDescription from "./FacebookDescription";

/**
 * Determines the width depending on the mode.
 *
 * @param {string} mode The mode. landscape, square, portrait.
 *
 * @returns {string} The width pixels.
 */
const determineTextContainerWidth = ( mode ) => {
	switch ( mode ) {
		case "landscape":
			return "527px";

		case "square":
			return "369px";

		case "portrait":
			return "369px";

		default:
			return "476px";
	}
};

const FacebookPreviewWrapper = styled.div`
	box-sizing: border-box;
	display: flex;
	flex-direction: ${ props => props.mode === "landscape" ? "column" : "row" };
	background-color: #f2f3f5;
	max-width: 527px;
`;

const FacebookTextWrapper = styled.div`
	box-sizing: border-box;
	background-color: #f2f3f5;
	margin: 0;
	padding: 10px 12px;
	position: relative;
	border-bottom: ${ props => props.mode === "landscape" ? "" : "1px solid #dddfe2" };
	border-top: ${ props => props.mode === "landscape" ? "" : "1px solid #dddfe2" };
	border-right: ${ props => props.mode === "landscape" ? "" : "1px solid #dddfe2" };
	border: ${ props => props.mode === "landscape" ? "1px solid #dddfe2" : "" };
	display: flex;
	flex-direction: column;
	flex-grow: 1;
	justify-content: ${ props => props.mode === "landscape" ? "flex-start" : "center" };
	font-size: 12px;
	overflow: hidden;
`;

/**
 * Renders FacebookPreview component.
 *
 * @param {object} props The props.
 *
 * @returns {React.Element} The rendered element.
 */
class FacebookPreview extends Component {
	/**
	 * The constructor.
	 *
	 * @param {Object} props The component's props.
	 */
	constructor( props ) {
		super( props );
		this.state = {
			imageMode: null,
		};
		this.onImageLoaded = this.onImageLoaded.bind( this );

		// Binding fields to onMouseHover to prevent arrow functions in JSX props.
		this.onImageEnter = this.props.onMouseHover.bind( this, "image" );
		this.onTitleEnter = this.props.onMouseHover.bind( this, "title" );
		this.onDescriptionEnter = this.props.onMouseHover.bind( this, "description" );
		this.onLeave = this.props.onMouseHover.bind( this, "" );

		// Binding fields to onSelect to prevent arrow functions in JSX props. Image field is handled in onImageClick function.
		this.onSelectTitle = this.props.onSelect.bind( this, "title" );
		this.onSelectDescription = this.props.onSelect.bind( this, "description" );
	}

	/**
	 * Retrieves the imageMode from the Facebook image container.
	 *
	 * @param {string} mode The Facebook image mode: landscape, portrait or square.
	 *
	 * @returns {void} Void.
	 */
	onImageLoaded( mode ) {
		this.setState( { imageMode: mode } );
	}

	/**
	 * Renders the FacebookPreview.
	 *
	 * @returns {ReactComponent} Either the PlaceholderImage component, the ErrorImage component or
	 * the TwitterImageContainer.
	 */
	render() {
		const { imageMode } = this.state;
		return (
			<FacebookPreviewWrapper
				id="facebookPreview"
				mode={ imageMode }
			>
				<FacebookImage
					src={ this.props.imageUrl || this.props.imageFallbackUrl }
					alt={ this.props.alt }
					onImageLoaded={ this.onImageLoaded }
					onImageClick={ this.props.onImageClick }
					onMouseEnter={ this.onImageEnter }
					onMouseLeave={ this.onLeave }
				/>
				<FacebookTextWrapper mode={ imageMode }>
					<FacebookSiteAndAuthorNames
						siteUrl={ this.props.siteUrl }
						authorName={ this.props.authorName }
						mode={ imageMode }
					/>
					<FacebookTitle
						onMouseEnter={ this.onTitleEnter }
						onMouseLeave={ this.onLeave }
						onClick={ this.onSelectTitle }
					>
						{ this.props.title }
					</FacebookTitle>
					<FacebookDescription
						maxWidth={ determineTextContainerWidth( imageMode ) }
						onMouseEnter={ this.onDescriptionEnter }
						onMouseLeave={ this.onLeave }
						onClick={ this.onSelectDescription }
						mode={ imageMode }
					>
						{ this.props.description }
					</FacebookDescription>
				</FacebookTextWrapper>
			</FacebookPreviewWrapper>
		);
	}
}

FacebookPreview.propTypes = {
	siteUrl: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
	authorName: PropTypes.string,
	description: PropTypes.string,
	imageUrl: PropTypes.string,
	imageFallbackUrl: PropTypes.string,
	alt: PropTypes.string,
	onSelect: PropTypes.func,
	onImageClick: PropTypes.func,
	onMouseHover: PropTypes.func,
};

FacebookPreview.defaultProps = {
	authorName: "",
	description: "",
	alt: "",
	imageUrl: "",
	imageFallbackUrl: "",
	onSelect: () => {},
	onImageClick: () => {},
	onMouseHover: () => {},
};

export default FacebookPreview;
