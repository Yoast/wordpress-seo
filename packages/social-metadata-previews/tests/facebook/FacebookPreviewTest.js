/* External dependencies */
import React from "react";
import renderer from "react-test-renderer";

/* Internal dependencies */
import FacebookPreview from "../../src/facebook/FacebookPreview";

describe( "FacebookPreview", () => {
	it( "matches the snapshot for a landscape image", () => {
		const component = renderer.create(
			<FacebookPreview
				siteUrl="yoast.com"
				title="YoastCon Workshops"
				description="Description to go along with a landscape image."
				src="https://yoast.com/app/uploads/2015/06/How_to_choose_keywords_FI.png"
				alt="Alt text"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
	it( "matches the snapshot for a portrait image", () => {
		const component = renderer.create(
			<FacebookPreview
				siteUrl="yoast.com"
				title="YoastCon Workshops"
				description="Description to go along with a portrait image."
				src="https://yoast.com/app/uploads/2015/09/Author_Joost_x2.png"
				alt="Alt text"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
	it( "matches the snapshot for a square image", () => {
		const component = renderer.create(
			<FacebookPreview
				siteUrl="yoast.com"
				title="YoastCon Workshops"
				description="Description to go along with a square image."
				src="https://yoast.com/app/uploads/2018/09/avatar_user_1_1537774226.png"
				alt="Alt text"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
	it( "matches the snapshot for a too small image", () => {
		const component = renderer.create(
			<FacebookPreview
				siteUrl="yoast.com"
				title="YoastCon Workshops"
				description="Description to go along with too small an image."
				src="https://yoast.com/app/uploads/2018/11/Logo_TYPO3-250x105.png"
				alt="Alt text"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
	it( "matches the snapshot for a faulty image", () => {
		const component = renderer.create(
			<FacebookPreview
				siteUrl="yoast.com"
				title="YoastCon Workshops"
				description="Description to go along with a faulty image."
				src="thisisnoimage"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
