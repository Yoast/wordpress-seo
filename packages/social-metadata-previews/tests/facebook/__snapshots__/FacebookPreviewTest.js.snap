// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookPreview matches the snapshot for a faulty image 1`] = `
.c3 {
  color: #606770;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 12px;
  line-height: 20px;
  overflow: hidden;
  padding: 0;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
  margin: 0;
  position: static;
}

.c4 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
}

.c1 {
  box-sizing: border-box;
  max-width: 527px;
  height: 273px;
  background-color: #ddd;
  border-style: dashed;
  border-width: 1px;
  color: #006DAC;
  background-color: #f1f1f1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-size: 14px;
  cursor: pointer;
}

.c5 {
  line-height: 20px;
  min-height: 20px;
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  margin: 0;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: pointer;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c6 {
  line-height: 16px;
  min-height: 16px;
  color: #606770;
  font-size: 12px;
  padding: 0;
  text-overflow: ellipsis;
  margin: 3px 0 0 0;
  display: -webkit-box;
  cursor: pointer;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: #f2f3f5;
  max-width: 527px;
}

.c2 {
  box-sizing: border-box;
  background-color: #f2f3f5;
  margin: 0;
  padding: 10px 12px;
  position: relative;
  border-bottom: 1px solid #dddfe2;
  border-top: 1px solid #dddfe2;
  border-right: 1px solid #dddfe2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
  overflow: hidden;
}

@media all and ( max-width:476px ) {
  .c6 {
    display: none;
  }
}

<div
  className="c0"
  id="facebookPreview"
  mode={null}
>
  <div
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Select image
  </div>
  <div
    className="c2"
    mode={null}
  >
    <span
      className="screen-reader-text"
    >
      yoast.com
    </span>
    <p
      aria-hidden="true"
      className="c3"
    >
      <span
        className="c4"
      >
        yoast.com
      </span>
    </p>
    <span
      className="c5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      YoastCon Workshops
    </span>
    <p
      className="c6"
      mode={null}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Description to go along with a faulty image.
    </p>
  </div>
</div>
`;

exports[`FacebookPreview matches the snapshot for a landscape image 1`] = `
.c3 {
  color: #606770;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 12px;
  line-height: 20px;
  overflow: hidden;
  padding: 0;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
  margin: 0;
  position: static;
}

.c4 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
}

.c1 {
  box-sizing: border-box;
  max-width: 527px;
  height: 273px;
  background-color: #ddd;
  border-style: dashed;
  border-width: 1px;
  color: #006DAC;
  background-color: #f1f1f1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-size: 14px;
  cursor: pointer;
}

.c5 {
  line-height: 20px;
  min-height: 20px;
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  margin: 0;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: pointer;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c6 {
  line-height: 16px;
  min-height: 16px;
  color: #606770;
  font-size: 12px;
  padding: 0;
  text-overflow: ellipsis;
  margin: 3px 0 0 0;
  display: -webkit-box;
  cursor: pointer;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: #f2f3f5;
  max-width: 527px;
}

.c2 {
  box-sizing: border-box;
  background-color: #f2f3f5;
  margin: 0;
  padding: 10px 12px;
  position: relative;
  border-bottom: 1px solid #dddfe2;
  border-top: 1px solid #dddfe2;
  border-right: 1px solid #dddfe2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
  overflow: hidden;
}

@media all and ( max-width:476px ) {
  .c6 {
    display: none;
  }
}

<div
  className="c0"
  id="facebookPreview"
  mode={null}
>
  <div
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Select image
  </div>
  <div
    className="c2"
    mode={null}
  >
    <span
      className="screen-reader-text"
    >
      yoast.com
    </span>
    <p
      aria-hidden="true"
      className="c3"
    >
      <span
        className="c4"
      >
        yoast.com
      </span>
    </p>
    <span
      className="c5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      YoastCon Workshops
    </span>
    <p
      className="c6"
      mode={null}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Description to go along with a landscape image.
    </p>
  </div>
</div>
`;

exports[`FacebookPreview matches the snapshot for a portrait image 1`] = `
.c3 {
  color: #606770;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 12px;
  line-height: 20px;
  overflow: hidden;
  padding: 0;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
  margin: 0;
  position: static;
}

.c4 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
}

.c1 {
  box-sizing: border-box;
  max-width: 527px;
  height: 273px;
  background-color: #ddd;
  border-style: dashed;
  border-width: 1px;
  color: #006DAC;
  background-color: #f1f1f1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-size: 14px;
  cursor: pointer;
}

.c5 {
  line-height: 20px;
  min-height: 20px;
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  margin: 0;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: pointer;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c6 {
  line-height: 16px;
  min-height: 16px;
  color: #606770;
  font-size: 12px;
  padding: 0;
  text-overflow: ellipsis;
  margin: 3px 0 0 0;
  display: -webkit-box;
  cursor: pointer;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: #f2f3f5;
  max-width: 527px;
}

.c2 {
  box-sizing: border-box;
  background-color: #f2f3f5;
  margin: 0;
  padding: 10px 12px;
  position: relative;
  border-bottom: 1px solid #dddfe2;
  border-top: 1px solid #dddfe2;
  border-right: 1px solid #dddfe2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
  overflow: hidden;
}

@media all and ( max-width:476px ) {
  .c6 {
    display: none;
  }
}

<div
  className="c0"
  id="facebookPreview"
  mode={null}
>
  <div
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Select image
  </div>
  <div
    className="c2"
    mode={null}
  >
    <span
      className="screen-reader-text"
    >
      yoast.com
    </span>
    <p
      aria-hidden="true"
      className="c3"
    >
      <span
        className="c4"
      >
        yoast.com
      </span>
    </p>
    <span
      className="c5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      YoastCon Workshops
    </span>
    <p
      className="c6"
      mode={null}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Description to go along with a portrait image.
    </p>
  </div>
</div>
`;

exports[`FacebookPreview matches the snapshot for a square image 1`] = `
.c3 {
  color: #606770;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 12px;
  line-height: 20px;
  overflow: hidden;
  padding: 0;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
  margin: 0;
  position: static;
}

.c4 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
}

.c1 {
  box-sizing: border-box;
  max-width: 527px;
  height: 273px;
  background-color: #ddd;
  border-style: dashed;
  border-width: 1px;
  color: #006DAC;
  background-color: #f1f1f1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-size: 14px;
  cursor: pointer;
}

.c5 {
  line-height: 20px;
  min-height: 20px;
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  margin: 0;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: pointer;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c6 {
  line-height: 16px;
  min-height: 16px;
  color: #606770;
  font-size: 12px;
  padding: 0;
  text-overflow: ellipsis;
  margin: 3px 0 0 0;
  display: -webkit-box;
  cursor: pointer;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: #f2f3f5;
  max-width: 527px;
}

.c2 {
  box-sizing: border-box;
  background-color: #f2f3f5;
  margin: 0;
  padding: 10px 12px;
  position: relative;
  border-bottom: 1px solid #dddfe2;
  border-top: 1px solid #dddfe2;
  border-right: 1px solid #dddfe2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
  overflow: hidden;
}

@media all and ( max-width:476px ) {
  .c6 {
    display: none;
  }
}

<div
  className="c0"
  id="facebookPreview"
  mode={null}
>
  <div
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Select image
  </div>
  <div
    className="c2"
    mode={null}
  >
    <span
      className="screen-reader-text"
    >
      yoast.com
    </span>
    <p
      aria-hidden="true"
      className="c3"
    >
      <span
        className="c4"
      >
        yoast.com
      </span>
    </p>
    <span
      className="c5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      YoastCon Workshops
    </span>
    <p
      className="c6"
      mode={null}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Description to go along with a square image.
    </p>
  </div>
</div>
`;

exports[`FacebookPreview matches the snapshot for a too small image 1`] = `
.c3 {
  color: #606770;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 12px;
  line-height: 20px;
  overflow: hidden;
  padding: 0;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
  margin: 0;
  position: static;
}

.c4 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
}

.c1 {
  box-sizing: border-box;
  max-width: 527px;
  height: 273px;
  background-color: #ddd;
  border-style: dashed;
  border-width: 1px;
  color: #006DAC;
  background-color: #f1f1f1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-size: 14px;
  cursor: pointer;
}

.c5 {
  line-height: 20px;
  min-height: 20px;
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  margin: 0;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: pointer;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c6 {
  line-height: 16px;
  min-height: 16px;
  color: #606770;
  font-size: 12px;
  padding: 0;
  text-overflow: ellipsis;
  margin: 3px 0 0 0;
  display: -webkit-box;
  cursor: pointer;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: #f2f3f5;
  max-width: 527px;
}

.c2 {
  box-sizing: border-box;
  background-color: #f2f3f5;
  margin: 0;
  padding: 10px 12px;
  position: relative;
  border-bottom: 1px solid #dddfe2;
  border-top: 1px solid #dddfe2;
  border-right: 1px solid #dddfe2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
  overflow: hidden;
}

@media all and ( max-width:476px ) {
  .c6 {
    display: none;
  }
}

<div
  className="c0"
  id="facebookPreview"
  mode={null}
>
  <div
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Select image
  </div>
  <div
    className="c2"
    mode={null}
  >
    <span
      className="screen-reader-text"
    >
      yoast.com
    </span>
    <p
      aria-hidden="true"
      className="c3"
    >
      <span
        className="c4"
      >
        yoast.com
      </span>
    </p>
    <span
      className="c5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      YoastCon Workshops
    </span>
    <p
      className="c6"
      mode={null}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Description to go along with too small an image.
    </p>
  </div>
</div>
`;
